### ISSUE TITLE
AR_STATUS_FATAL when creating session in ARCore 1.5 on OnePlus A5000

### ISSUE THREAD

**Issue by Tatsi** [google-ar/arcore-android-sdk#595]
In example application *hello_ar_c*, creating a session produces AR_ERROR_FATAL when using ARCore SDK 1.5 and included ARCore apk or one the latest from Play Store (1.5). 
```
CHECK(ArSession_create(env, context, &ar_session_) == AR_SUCCESS);
```

I've tested all combinations of SDK examples and ARCore *apk*'s to make sure the problem appeared in 1.5:

- ARCore APK 1.1 & 1.1 SDK hello_ar_c = works
- ARCore APK 1.2 & 1.2 SDK hello_ar_c = works
- ARCore APK 1.2 & 1.2.1 SDK hello_ar_c = works
- ARCore APK 1.3 & 1.3 SDK hello_ar_c = works
- ARCore APK 1.4 & 1.4 SDK hello_ar_c = works
- ARCore APK 1.5 & 1.5 SDK hello_ar_c = **AR_ERROR_FATAL**
- ARCore APK from Play Store (1.5)  & 1.5 SDK hello_ar_c = **AR_ERROR_FATAL**

### Device infromation:
_VERSION.RELEASE : 8.1.0
VERSION.INCREMENTAL : 994cc25e29
VERSION.SDK.NUMBER : 27
BOARD : msm8998
BOOTLOADER : unknown
BRAND : OnePlus
DISPLAY : lineage_cheeseburger-userdebug 8.1.0 OPM1.171019.018 994cc25e29
FINGERPRINT : OnePlus/OnePlus5/OnePlus5:8.0.0/OPR6.170623.013/01132342:user/release-keys
HARDWARE : qcom
HOST : agrippa.acc.umu.se
ID : OPM1.171019.018
MANUFACTURER : OnePlus
MODEL : ONEPLUS A5000
PRODUCT : OnePlus5
SERIAL : d664dfc5
TAGS : release-keys
TIME : 1519635365000_

**Comment by inio** [comment]
Could I get a logcat dump of anything printed during `ArSession_create`?

**Comment by Tatsi** [comment]
```
LOGI("#########################");
LOGI("Calling ArSession_create");
ArStatus createStatus = ArSession_create(env, context, &ar_session_);
LOGI("Called ArSession_create");
LOGI("#########################");
CHECK(createStatus == AR_SUCCESS);
```

Produces this:

```
2018-10-07 14:29:48.227 5568-5568/com.google.ar.core.examples.c.helloar I/hello_ar_example_c: #########################
2018-10-07 14:29:48.227 5568-5568/com.google.ar.core.examples.c.helloar I/hello_ar_example_c: Calling ArSession_create
2018-10-07 14:29:48.227 5568-5568/com.google.ar.core.examples.c.helloar I/third_party/arcore/ar/core/android/sdk/session_create.cc: Entering ArSession_create
2018-10-07 14:29:48.235 5568-5568/com.google.ar.core.examples.c.helloar I/third_party/arcore/ar/core/android/sdk/session_create.cc: ARCore Version: APK version code: 180910056
2018-10-07 14:29:48.982 2561-2725/? I/WCNSS_FILTER: ibs_msm_serial_clock_vote: vote UART CLK ON using UART driver's ioctl()
2018-10-07 14:29:48.985 2561-2725/? E/WCNSS_FILTER: wcnss_acquire_wakelock wake lock not initialized
2018-10-07 14:29:48.985 2561-2725/? I/WCNSS_FILTER: ibs_bt_device_wakeup: Writing IBS_WAKE_IND
2018-10-07 14:29:48.985 2561-2725/? I/WCNSS_FILTER: do_write: IBS write: fd
2018-10-07 14:29:48.989 2561-2561/? I/WCNSS_FILTER: ibs_recv_ibs_cmd: Received IBS_WAKE_ACK: 0xFC
2018-10-07 14:29:48.991 2561-2561/? I/WCNSS_FILTER: ibs_recv_ibs_cmd: Received IBS_WAKE_IND: 0xFD
2018-10-07 14:29:48.991 2561-2561/? I/WCNSS_FILTER: ibs_recv_ibs_cmd: Writing IBS_WAKE_ACK
2018-10-07 14:29:48.991 2561-2561/? I/WCNSS_FILTER: do_write: IBS write: fc
2018-10-07 14:29:49.053 2561-2561/? I/WCNSS_FILTER: ibs_recv_ibs_cmd: Received IBS_SLEEP_IND: 0xFE
2018-10-07 14:29:49.105 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_GNSS_MEASUREMENT_REPORT_IND_V02
2018-10-07 14:29:49.105 942-2244/? I/LocSvc_ApiV02: [SvMeas] SeqNum: 1, MaxMsgNum: 4, MeasValid: 0, #of SV: 0
2018-10-07 14:29:49.105 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3505]: there is no valid SV measurements
2018-10-07 14:29:49.105 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3524]: There is no GNSS measurement.
2018-10-07 14:29:49.105 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_GNSS_MEASUREMENT_REPORT_IND_V02
2018-10-07 14:29:49.106 942-2244/? I/LocSvc_ApiV02: [SvMeas] SeqNum: 2, MaxMsgNum: 4, MeasValid: 0, #of SV: 0
2018-10-07 14:29:49.106 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3505]: there is no valid SV measurements
2018-10-07 14:29:49.106 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3524]: There is no GNSS measurement.
2018-10-07 14:29:49.106 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_GNSS_MEASUREMENT_REPORT_IND_V02
2018-10-07 14:29:49.106 942-2244/? I/LocSvc_ApiV02: [SvMeas] SeqNum: 3, MaxMsgNum: 4, MeasValid: 0, #of SV: 0
2018-10-07 14:29:49.106 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3505]: there is no valid SV measurements
2018-10-07 14:29:49.106 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3524]: There is no GNSS measurement.
2018-10-07 14:29:49.106 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_GNSS_MEASUREMENT_REPORT_IND_V02
2018-10-07 14:29:49.106 942-2244/? I/LocSvc_ApiV02: [SvMeas] SeqNum: 4, MaxMsgNum: 4, MeasValid: 0, #of SV: 0
2018-10-07 14:29:49.106 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3505]: there is no valid SV measurements
2018-10-07 14:29:49.106 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3524]: There is no GNSS measurement.
2018-10-07 14:29:49.107 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_GNSS_SV_INFO_IND_V02
2018-10-07 14:29:49.107 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_POSITION_REPORT_IND_V02
2018-10-07 14:29:49.107 942-3215/? I/LocSvc_libulp: ulp_brain_process_gnss_sv_report, gnss sv report cnt = 1344, gnss pos report cnt = 155,strong sv cnt = 0
2018-10-07 14:29:49.107 942-3215/? I/LocSvc_libulp: ulp_brain_process_gnss_position_report, transition provider returned = 0
2018-10-07 14:29:49.107 942-3215/? I/LocSvc_libulp: ulp_brain_process_gnss_position_report, report GNSS position/status to hybrid fix requests = 0
2018-10-07 14:29:49.224 5568-5568/com.google.ar.core.examples.c.helloar W/native: commandlineflags.cc:943 Ignoring RegisterValidateFunction() for flag pointer 0x73c0467830: no flag found at that address
2018-10-07 14:29:49.225 5568-5568/com.google.ar.core.examples.c.helloar I/native: version_check.cc:23 ARCore Version: APK version:1.5.180910056
2018-10-07 14:29:49.225 5568-5568/com.google.ar.core.examples.c.helloar I/native: version_check.cc:24 ARCore Version: emulated SDK version:1.5.180815000
2018-10-07 14:29:49.225 5568-5568/com.google.ar.core.examples.c.helloar I/third_party/arcore/ar/core/android/sdk/session_create.cc: ARCore Version: SDK build name: 1.5
2018-10-07 14:29:49.225 5568-5568/com.google.ar.core.examples.c.helloar I/third_party/arcore/ar/core/android/sdk/session_create.cc: Dynamite load ok.
2018-10-07 14:29:49.547 1193-1245/system_process W/ActivityManager: Launch timeout has expired, giving up wake lock!
2018-10-07 14:29:49.924 5568-5568/com.google.ar.core.examples.c.helloar W/native: commandlineflags.cc:943 Ignoring RegisterValidateFunction() for flag pointer 0x73c0388ec0: no flag found at that address
2018-10-07 14:29:49.932 5568-5568/com.google.ar.core.examples.c.helloar I/native: session_create_implementation.cc:47 Entering ArSession_createWithSettingsImplementation_private. ARCore SDK version: [1.5.180815000].
2018-10-07 14:29:49.959 31901-32147/com.google.ar.core W/native: helpers.cc:1599 Defaulting to persistent calibration file.
2018-10-07 14:29:49.959 31901-32147/com.google.ar.core W/native: helpers.cc:1576 Property calibration_cad is not defined.
2018-10-07 14:29:49.998 5568-5568/com.google.ar.core.examples.c.helloar I/native: device_profile_conversions.cc:142 used_active_calibration: 2
2018-10-07 14:29:50.001 5568-5568/com.google.ar.core.examples.c.helloar I/native: device_profile_conversions.cc:142 used_active_calibration: 2
2018-10-07 14:29:50.007 5568-5568/com.google.ar.core.examples.c.helloar I/native: android_sensors.cc:80 Using uncalibrated accelerometer.
2018-10-07 14:29:50.008 5568-5568/com.google.ar.core.examples.c.helloar I/native: android_sensors.cc:110 Using uncalibrated magnetometer.
2018-10-07 14:29:50.012 2561-2737/? I/WCNSS_FILTER: ibs_wcnss_bt_device_sleep: TX Awake, Sending SLEEP_IND
2018-10-07 14:29:50.012 2561-2737/? I/WCNSS_FILTER: do_write: IBS write: fe
2018-10-07 14:29:50.012 2561-2737/? I/WCNSS_FILTER: ibs_msm_serial_clock_vote: vote UART CLK OFF using UART driver's ioctl()
2018-10-07 14:29:50.012 2561-2737/? E/WCNSS_FILTER: wcnss_release_wakelock wake lock not initialized
2018-10-07 14:29:50.017 5568-5568/com.google.ar.core.examples.c.helloar I/native: performance_manager.cc:22 Processor data: Num processors 8
    Processor type: LittleBig
    Num small cores 4
    Max frequency 1900800
    Min frequency 300000
    Small cores 0 1 2 3
    Num big cores 4
    Max frequency 2457600
    Min frequency 300000
    Big cores 4 5 6 7
2018-10-07 14:29:50.034 5568-5568/com.google.ar.core.examples.c.helloar I/native: device_provider.cc:1177 Identified device type: kOneplus5
2018-10-07 14:29:50.069 5568-5568/com.google.ar.core.examples.c.helloar I/native: vio_helper.cc:49 Use calibrated IMU scale factors.
2018-10-07 14:29:50.069 5568-5568/com.google.ar.core.examples.c.helloar I/native: vio_helper.cc:61 Use calibrated IMU misalignment factors.
2018-10-07 14:29:50.103 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_GNSS_MEASUREMENT_REPORT_IND_V02
2018-10-07 14:29:50.103 942-2244/? I/LocSvc_ApiV02: [SvMeas] SeqNum: 1, MaxMsgNum: 4, MeasValid: 0, #of SV: 0
2018-10-07 14:29:50.103 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3505]: there is no valid SV measurements
2018-10-07 14:29:50.103 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3524]: There is no GNSS measurement.
2018-10-07 14:29:50.103 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_GNSS_MEASUREMENT_REPORT_IND_V02
2018-10-07 14:29:50.103 942-2244/? I/LocSvc_ApiV02: [SvMeas] SeqNum: 2, MaxMsgNum: 4, MeasValid: 0, #of SV: 0
2018-10-07 14:29:50.103 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3505]: there is no valid SV measurements
2018-10-07 14:29:50.103 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3524]: There is no GNSS measurement.
2018-10-07 14:29:50.103 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_GNSS_MEASUREMENT_REPORT_IND_V02
2018-10-07 14:29:50.103 942-2244/? I/LocSvc_ApiV02: [SvMeas] SeqNum: 3, MaxMsgNum: 4, MeasValid: 0, #of SV: 0
2018-10-07 14:29:50.103 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3505]: there is no valid SV measurements
2018-10-07 14:29:50.103 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3524]: There is no GNSS measurement.
2018-10-07 14:29:50.103 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_GNSS_MEASUREMENT_REPORT_IND_V02
2018-10-07 14:29:50.103 942-2244/? I/LocSvc_ApiV02: [SvMeas] SeqNum: 4, MaxMsgNum: 4, MeasValid: 0, #of SV: 0
2018-10-07 14:29:50.103 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3505]: there is no valid SV measurements
2018-10-07 14:29:50.103 942-2244/? E/LocSvc_ApiV02: reportGnssMeasurementData:3524]: There is no GNSS measurement.
2018-10-07 14:29:50.104 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_GNSS_SV_INFO_IND_V02
2018-10-07 14:29:50.104 942-2244/? I/LocSvc_ApiV02: <--- globalEventCb line 115 QMI_LOC_EVENT_POSITION_REPORT_IND_V02
2018-10-07 14:29:50.104 942-3215/? I/LocSvc_libulp: ulp_brain_process_gnss_sv_report, gnss sv report cnt = 1345, gnss pos report cnt = 155,strong sv cnt = 0
2018-10-07 14:29:50.104 942-3215/? I/LocSvc_libulp: ulp_brain_process_gnss_position_report, transition provider returned = 0
2018-10-07 14:29:50.104 942-3215/? I/LocSvc_libulp: ulp_brain_process_gnss_position_report, report GNSS position/status to hybrid fix requests = 0
2018-10-07 14:29:50.145 5568-5568/com.google.ar.core.examples.c.helloar I/native: device_profile_conversions.cc:142 used_active_calibration: 2
2018-10-07 14:29:50.155 5568-5568/com.google.ar.core.examples.c.helloar W/AnchorServiceClientFactory: The API key for use with the Google AR service could not be obtained!
2018-10-07 14:29:50.180 5568-5568/com.google.ar.core.examples.c.helloar I/native: android_camera.cc:1187 Initializing camera manager.
2018-10-07 14:29:50.190 5568-5568/com.google.ar.core.examples.c.helloar I/native: android_camera.cc:1211 Camera manager initialized successfully with 3 cameras.
2018-10-07 14:29:50.190 5568-5568/com.google.ar.core.examples.c.helloar E/native: status.cc:155 generic::internal: Physical cameras don't match device profile. 1 expected back facing:0 actual back facing:1
2018-10-07 14:29:50.190 5568-5568/com.google.ar.core.examples.c.helloar E/native: session_create_implementation.cc:183 Failed to initilize the data source - -2
2018-10-07 14:29:50.190 5568-5568/com.google.ar.core.examples.c.helloar E/native: session_create_implementation.cc:184 ArSession_createWithSettingsImplementation_private returning ArStatusErrorSpace::AR_ERROR_FATAL: .
2018-10-07 14:29:50.200 5568-5568/com.google.ar.core.examples.c.helloar I/native: feature_combo_node.cc:196 Cleared scoped_image buffers in feature_combo.
2018-10-07 14:29:50.246 5568-5568/com.google.ar.core.examples.c.helloar I/native: android_camera.cc:696 Reset cleanly got to CLOSED state for camera 0
2018-10-07 14:29:50.248 5568-5568/com.google.ar.core.examples.c.helloar I/native: android_camera.cc:696 Reset cleanly got to CLOSED state for camera 2
2018-10-07 14:29:50.251 5568-5568/com.google.ar.core.examples.c.helloar I/native: android_camera.cc:696 Reset cleanly got to CLOSED state for camera 1
2018-10-07 14:29:50.254 5568-5568/com.google.ar.core.examples.c.helloar I/hello_ar_example_c: Called ArSession_create
2018-10-07 14:29:50.254 5568-5568/com.google.ar.core.examples.c.helloar I/hello_ar_example_c: #########################
```

**Comment by inio** [comment]
Well that's a new one:

    I/native: android_camera.cc:1211 Camera manager initialized successfully with 3 cameras.
    E/native: status.cc:155 generic::internal: Physical cameras don't match device profile. 1 expected back 
    E/native: session_create_implementation.cc:183 Failed to initilize the data source - -2

I'll get this to our device certification group and see what they can make of it.

**Comment by blackarrow13** [comment]
I have the same issue on Sony Xperia XZ3, using ArCore 1.14.
Any news there?

**Comment by blackarrow13** [comment]
The problem was solved for me in ArCore 1.15

**Comment by devbridie** [comment]
As a part of ongoing efforts to maintain ARCore repositories, I’m closing this issue as stale since it hasn’t had activity in more than 6 months. If this issue is still relevant, please feel free to file a new bug using the latest version of the SDK.
